### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by kifle23**
In this PR, I have done the following:

- [x] Used Redux Toolkit
- [x] Created a directory (/src/redux)
- [x] Configured a Redux store (/src/redux/store.js)
- [x] Defined a slice of state for books with an initial state: empty array
- [x] Included a reducer that adds a book
- [x] Included a reducer that removes a book
- [x] Defined a slice of state for categories with an initial state: empty array
- [x] Included a reducer that checks the status and always returns "Under construction"

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      return [...state, action.payload];
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.payload.id);
+    default:
+      return state;
+  }
+}
+
+export const addBook = (book) => ({ type: ADD_BOOK, payload: book });
+export const removeBook = (book) => ({ type: REMOVE_BOOK, payload: book });
```
Comment (OyePriscilla) [inline src/redux/books/booksSlice.js]: - [x] Kindly take note that you can create a slice using Redux Toolkit's createSlice function in your React application accoring to the project requirement. This approach simplifies the setup and reduces the boilerplate required for Redux actions and reducers.

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @kifle23 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Use store, actions, and reducers in React ✔️ 
Adding and removing books functionalities worked as expected ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (kifle23) [reply src/redux/books/booksSlice.js]: Thank you for your review @OyePriscilla

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
+    removeBook: (state, action) => state.filter((book) => book.id !== action.payload.id),
+  },
```
Comment (binhussen) [inline src/redux/books/booksSlice.js]: - [Optional] Kindly `filter()` method does not mutate the original array but instead returns a new array populated with elements of filtered books, so here you are not assigning the returned value to the book state, please assign it to not lose what you removed 👍 
`state.books = state.filter((book) => book.id !== action.payload.id)`

**Review by binhussen — APPROVED** [review]
Hi @kifle23 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Used toolkit & react-redux ✔️
- Prepared slices ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 2 above)
Comment (kifle23) [reply src/redux/books/booksSlice.js]: Thank you @binhussen; I will incorporate the changes!

Comment (kifle23) [reply src/redux/books/booksSlice.js]: Hi @binhussen, I have tried to work on your suggestion if you have any feedback please let me know 🙏

**Pull request merged**